src/kvstore/plugins/elasticsearch/ESListener.cpp

```diff
@@ -0,0 +1,272 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "utils/NebulaKeyUtils.h"
+#include "kvstore/plugins/elasticsearch/ESListener.h"
+#include "common/plugin/fulltext/elasticsearch/ESStorageAdapter.h"
+
+DECLARE_int32(ft_request_retry_times);
+
+namespace nebula {
+namespace kvstore {
+bool ESListener::apply(const std::vector<KV>& data) {
+    std::vector<nebula::plugin::DocItem> docItems;
+    auto vIdLen = getVidLen();
+    if (!vIdLen.ok()) {
+        return false;
+    }
+    for (const auto& kv : data) {
```

Comment:
Maybe we need to split into batch if necessary?

Reply:
> Maybe we need to split into batch if necessary?

Yeah, solved.